### ISSUE TITLE
"node" and "npm": executable file not found in $PATH

### ISSUE THREAD

**Issue by ignaciomella** [markshust/docker-magento#694]
**Description**
I need to use npm and nodejs, but it seems that they are not installed. 

**Steps To Reproduce**
1. New fresh installation through automated one-liner
2. _exectute bin/npm --version_  and _bin/node --version_

**Expected Result**
No erros. 

Error: _Executable file not found in $PATH_

**Comment by TerrorSquad** [comment]
I just came here to report the same thing. The latest image doesn't have node and npm in the PATH.
The previous version work though - markoshust/magento-php:7.4-fpm-13

**Comment by magedevjosh** [comment]
I ran into this as well. Ended up having to bash into the container and use [this link] (https://www.rosehosting.com/blog/how-to-install-node-js-and-npm-on-debian-11/) to install node/npm. Then, you'll need to update the cli commands on your local to run the command as npx instead of npm.

**Comment by dirkreitz** [comment]
obviously the new 'streamlined' php images don't have node support at all - I do not know if this is on purpose or by accident

**Comment by dirkreitz** [comment]
As I build my own php images, I still install node with the following lines:

RUN apt-get install -y gnupg \
  && curl -sL https://deb.nodesource.com/setup_current.x | bash - \
  && apt-get install -y nodejs \
  && mkdir /var/www/.config /var/www/.npm \
  && chown app:app /var/www/.config /var/www/.npm \
  && npm install -g grunt-cli

Mark had the same setup, but was using an outdated version of node

**Comment by ignaciomella** [comment]
The thing is... there is no reference to this _strimlined_ image being not compatible with the custom CLI Commands. 

If so, it would be nice to reference this lack of support somewhere in the README.md

**Comment by dirkreitz** [comment]
I would like to give Mark a chance to answer, why he dismissed the installation of node.
As far as I know, php:7.4-fpm-13 should work ...

**Comment by Viper9x** [comment]
I confirmed that npm and node does not available on the latest Automated Setup which has markoshust/magento-php:8.1-fpm-0

**Comment by markshust** [comment]
Sorry for the delay in my response of this issue, just got over a big course creation bender.

I forgot to note the removal of node & npm in the latest images. If you absolutely need these, I'd recommend reverting back to the previous Docker images for the time being.

I'd like to move node/npm to their own containers. I wasn't sure if the code is dependent upon the PHP images or not, but there should be a way to make the binaries accessible within their own containers. This would provide the ability to update node/npm sources without affecting the PHP image.

I'll try to take a look at this soon, but if anyone wants to take a stab at this, let me know!

**Comment by artmouse** [comment]
> 
> Or create a "bin/setup-grunt" script. Would you have an idea ? When I ran the first command. I had an error : `bin/cli apt install nodejs npm`
> 
> `E: Could not open lock file /var/lib/dpkg/lock-frontend - open (13: Permission denied)` `E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), are you root?`
> 
> Thanks for all,

```
bin/root apt update
bin/root apt install nodejs npm
cd src
cp Gruntfile.js.sample Gruntfile.js
cp grunt-config.json.sample grunt-config.json
cp package.json.sample package.json
cd ..
bin/root npm install -g grunt-cli
bin/npm install
bin/npm update
```

**Comment by malou72** [comment]
Thanks @artmouse for your reply !

I choosed to downgrade `markoshust/magento-php:7.4-fpm-13` and It works.

**Comment by drtrem86** [comment]
> [use](/artmouse)
`bin/rootnotty`
`bin/rootnotty apt install nodejs npm`
standalone container

**Comment by thinkfix** [comment]
I use markoshust/magento-php:8.1-fpm-develop
node --version
v14.19.1
npm --version
6.14.16

**Comment by markshust** [comment]
I was planning on putting npm & node in their own container, but it seems there is a lot of dependencies such as cache-clean not working unless node is installed. I'll be adding them back into this main image momentarily and will rebuild these images 👍 Really sorry if this has inconvenienced anyone, but will get these packages added very soon.

**Comment by markshust** [comment]
I did forget there is an open PR which could potentially add this as a separate container instead https://github.com/markshust/docker-magento/pull/699/files .... looking at this before rebuilding images.

**Comment by markshust** [comment]
Rebuilding PHP images/tags https://github.com/markshust/docker-magento/actions/runs/2387076836

**Comment by markshust** [comment]
Released in 43.0.0
